cmd/provisioner-localpv/app/provisioner.go

```diff
@@ -114,68 +113,14 @@ func (p *Provisioner) Provision(opts pvController.VolumeOptions) (*v1.Persistent
 
 	//TODO: Determine if hostpath or device based Local PV should be created
 	stgType := pvCASConfig.GetStorageType()
-	if stgType != "hostpath" {
-		return nil, fmt.Errorf("PV with StorageType %v is not supported", stgType)
+	if stgType == "hostpath" {
+		return p.ProvisionHostPath(opts, pvCASConfig)
 	}
-
-	path, err := pvCASConfig.GetPath()
-	if err != nil {
-		return nil, err
-	}
-
-	glog.Infof("Creating volume %v at %v:%v", name, node.Name, path)
-
-	//Before using the path for local PV, make sure it is created.
-	initCmdsForPath := []string{"mkdir", "-m", "0777", "-p"}
-	podOpts := &HelperPodOptions{
-		cmdsForPath: initCmdsForPath,
-		name:        name,
-		path:        path,
-		nodeName:    node.Name,
-	}
-
-	iErr := p.createInitPod(podOpts)
-	if iErr != nil {
-		glog.Infof("Initialize volume %v failed: %v", name, iErr)
-		return nil, iErr
-	}
-
-	// VolumeMode will always be specified as Filesystem for host path volume,
-	// and the value passed in from the PVC spec will be ignored.
-	fs := v1.PersistentVolumeFilesystem
-
-	// It is possible that the HostPath doesn't already exist on the node.
-	// Set the Local PV to create it.
-	//hostPathType := v1.HostPathDirectoryOrCreate
-
-	// TODO initialize the Labels and annotations
-	// Use annotations to specify the context using which the PV was created.
-	//volAnnotations := make(map[string]string)
-	//volAnnotations[string(v1alpha1.CASTypeKey)] = casVolume.Spec.CasType
-	//fstype := casVolume.Spec.FSType
-
-	labels := make(map[string]string)
-	labels[string(mconfig.CASTypeKey)] = "local-" + stgType
-	//labels[string(v1alpha1.StorageClassKey)] = *className
-
-	//TODO Change the following to a builder pattern
-	pvObj, err := mPV.NewBuilder().
-		WithName(name).
-		WithLabels(labels).
-		WithReclaimPolicy(opts.PersistentVolumeReclaimPolicy).
-		WithAccessModes(pvc.Spec.AccessModes).
-		WithVolumeMode(fs).
-		WithCapacityQty(pvc.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]).
-		WithLocalHostDirectory(path).
-		WithNodeAffinity(node.Name).
-		Build()
-
-	if err != nil {
-		return nil, err
+	if stgType == "device" {
+		return p.ProvisionBlockDevice(opts, pvCASConfig)
```

Comment:
nit : both if cases are returning same method

Reply:
It is between BlockDevice and Hostpath.